outlines/security.md

```diff
@@ -0,0 +1,55 @@
+# Security
+
+1. Main concept: Security surface area of a Meteor app
+    1. The way to security is to understand the attack surface, and guard against all possible attacks
+    2. Conclusion: Never Trust The Client - anyone can call all of your app's endpoints, not just the client you wrote. There's no way to guard against this, so you may as well not even try. Consider projects like SnapchatFS
+    3. It's not always clear where the server ends and the client begins in a Meteor app, but it's important to be aware of that for security
+2. Methods
+    1. Don't write generic methods, make sure you know what each argument is exactly, and what it could do
+    2. Rate limiting as a first line of defense against brute force
+    3. Make sure method side effects don't give away information. For example, returning how many items were affected should only tell the user about documents they should be able to access
+    4. Use this.userId, never take the current user as an argument in a method
+    5. Always include the user ID and any prerequisites directly in the query, in case that stuff changed since you did the security check
```

Comment:
Improved.

Changed to: "Always include the user ID and any prerequisites in the selector part of every update query, in case that stuff changed since you did the security check. This is the only way to do "atomic" updates in Mongo."